src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests5.cs

```diff
@@ -315,50 +316,671 @@ public static void Main()
+            var isPattern = tree.GetRoot().DescendantNodes().OfType<IsPatternExpressionSyntax>().Single();
+
+            VerifyOperationTree(comp, model.GetOperation(isPattern), @"
+IIsPatternOperation (OperationKind.IsPattern, Type: System.Boolean, IsInvalid) (Syntax: 'this is { M ... ng2: null }')
+  Value:
+    IInstanceReferenceOperation (ReferenceKind: ContainingTypeInstance) (OperationKind.InstanceReference, Type: C) (Syntax: 'this')
+  Pattern:
+    IRecursivePatternOperation (OperationKind.RecursivePattern, Type: null, IsInvalid) (Syntax: '{ Missing1. ... ng2: null }') (InputType: C, NarrowedType: C, DeclaredSymbol: null, MatchedType: C, DeconstructSymbol: null)
+      DeconstructionSubpatterns (0)
+      PropertySubpatterns (1):
+          IPropertySubpatternOperation (OperationKind.PropertySubpattern, Type: null, IsInvalid, IsImplicit) (Syntax: 'Missing1')
+            Member:
+              IInvalidOperation (OperationKind.Invalid, Type: null, IsInvalid) (Syntax: 'Missing1')
+                Children(0)
+            Pattern:
+              IRecursivePatternOperation (OperationKind.RecursivePattern, Type: null, IsInvalid, IsImplicit) (Syntax: 'Missing1') (InputType: ?, NarrowedType: ?, DeclaredSymbol: null, MatchedType: ?, DeconstructSymbol: null)
+                DeconstructionSubpatterns (0)
+                PropertySubpatterns (1):
+                    IPropertySubpatternOperation (OperationKind.PropertySubpattern, Type: null, IsInvalid, IsImplicit) (Syntax: 'Missing1.Missing2')
+                      Member:
+                        IInvalidOperation (OperationKind.Invalid, Type: null, IsInvalid) (Syntax: 'Missing1.Missing2')
+                          Children(0)
+                      Pattern:
+                        IConstantPatternOperation (OperationKind.ConstantPattern, Type: null) (Syntax: 'null') (InputType: ?, NarrowedType: ?)
+                          Value:
+                            IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: ?, Constant: null, IsImplicit) (Syntax: 'null')
+                              Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: True, IsUserDefined: False) (MethodSymbol: null)
+                              Operand:
+                                ILiteralOperation (OperationKind.Literal, Type: null, Constant: null) (Syntax: 'null')
+");
+        }
+
+        [Fact, WorkItem(53484, "https://github.com/dotnet/roslyn/issues/53484")]
+        public void ExtendedPropertyPatterns_SuppressionOnPattern()
+        {
+            var program = @"
+#nullable enable
+public class ContainerType
+{
+    public class Type
+    {
+        public void M()
+        {
+            const Type c = null!;
+            if (this is c!) {}
+            if (this is (c!)) {}
+            if (this is Type!) {} // 1
+            if (this is ContainerType!.Type) {} // 2
+            if (this is ContainerType.Type!) {} // 3
```

Comment:
@jcouv Why is it expected to accept suppression in the first two?

Reply:
Those already compile.

Reply:
I get parsing errors on sharplab default branch. 

https://sharplab.io/#v2:D4AQTAjAsAUAxAOwK4BsUEMBGKCmACHBLXWEAZj3DwGE8BvWPJyikAFjwFkAKASnsbMhAYwD2CAM4AXGnmF4AvHmRoA3IKFMA+orxSAFgEsJeY3ICE6mEIC+sG0A

I think patterns did not accept suppression in the previous version of the compiler.

Reply:
Oops, I used sharplab on `main`, which already has this change...
Filed https://github.com/dotnet/roslyn/issues/54071